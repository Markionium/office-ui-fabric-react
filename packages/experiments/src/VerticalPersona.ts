export * from './components/VerticalPersona/index';
