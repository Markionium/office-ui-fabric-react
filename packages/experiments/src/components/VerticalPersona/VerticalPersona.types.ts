import { IStyle, IFontWeight } from 'office-ui-fabric-react/lib/Styling';
import { IStatelessComponent, IStyleableComponentProps } from '../../Foundation';
import { IPersonaCoinProps } from '../PersonaCoin/PersonaCoin.types';

export type IVerticalPersonaComponent = IStatelessComponent<IVerticalPersonaProps, IVerticalPersonaStyles>;

// Extending IStyleableComponentProps will automatically add stylable props for you, such as styles and theme.
//    If you don't want these props to be included in your component, just remove this extension.
export interface IVerticalPersonaProps extends IStyleableComponentProps<IVerticalPersonaProps, IVerticalPersonaStyles> {
  text: string;
  secondaryText?: string;

  /**
   * Properties that should be passed to the PersonaCoin component
   */
  coinProps?: Partial<IPersonaCoinProps>;
  styleVariables?: IVerticalPersonaStyleVariableTypes;
}

export interface IVerticalPersonaStyles {
  root: IStyle;
  text: IStyle;
  secondaryText: IStyle;
}

export interface IVerticalPersonaStyleVariableTypes {
  text: ITextStyleVariables;
  primaryText: IPrimaryTextStyleVariables;
  secondaryText: ISecondaryTextStyleVariables;
}

interface ITextStyleVariables {
  height: string;
  fontFamily: string;
  width: string;
}

interface IPrimaryTextStyleVariables {
  paddingTop: string;
  fontSize: string;
  fontWeight: IFontWeight;
}

interface ISecondaryTextStyleVariables {
  paddingTop: string;
  fontSize: string;
}
