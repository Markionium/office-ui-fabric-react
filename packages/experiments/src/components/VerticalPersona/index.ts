export * from './VerticalPersona';
export * from './VerticalPersona.types';
