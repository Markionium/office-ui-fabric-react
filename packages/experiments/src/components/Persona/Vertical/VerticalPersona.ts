import { VerticalPersonaView } from './VerticalPersona.view';
import { VerticalPersonaStyles, VerticalPersonaTokens } from './VerticalPersona.styles';
import { IVerticalPersonaProps, IVerticalPersonaStyles, IVerticalPersonaTokens } from './VerticalPersona.types';
import { createStatelessComponent } from '../../../Foundation';

export const VerticalPersona: React.StatelessComponent<IVerticalPersonaProps> = createStatelessComponent<
  IVerticalPersonaProps,
  IVerticalPersonaStyles,
  IVerticalPersonaTokens
>({
  displayName: 'VerticalPersona',
  view: VerticalPersonaView,
  styles: VerticalPersonaStyles,
  tokens: VerticalPersonaTokens
});
