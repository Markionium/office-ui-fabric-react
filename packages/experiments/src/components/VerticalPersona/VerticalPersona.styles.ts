import { IVerticalPersonaComponent } from './VerticalPersona.types';
import { processVariables } from '../../utilities/VariableProcessing';

export const VerticalPersonaStyles: IVerticalPersonaComponent['styles'] = props => {
  const { theme } = props;

  const variables = processVariables(
    {
      text: {
        height: '35px',
        fontFamily: 'SegoeUI',
        width: '106px'
      },
      primaryText: {
        paddingTop: '8px',
        fontSize: '14px',
        fontWeight: 600
      },
      secondaryText: {
        paddingTop: '12px',
        fontSize: '12px'
      }
    },
    props.styleVariables
  );

  return {
    root: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      justifyContent: 'flex-start',
      width: '122px'
    },
    text: {
      paddingTop: variables.primaryText.paddingTop,
      width: variables.text.width,
      height: variables.text.height,
      fontFamily: variables.text.fontFamily,
      fontSize: variables.primaryText.fontSize,
      fontWeight: variables.primaryText.fontWeight,
      color: theme.palette.neutralPrimary
    },
    secondaryText: {
      paddingTop: variables.secondaryText.paddingTop,
      width: variables.text.width,
      height: variables.text.height,
      fontFamily: variables.text.fontFamily,
      fontSize: variables.secondaryText.fontSize,
      lineHeight: '1.42',
      textAlign: 'center',
      color: theme.palette.neutralSecondary
    }
  };
};
