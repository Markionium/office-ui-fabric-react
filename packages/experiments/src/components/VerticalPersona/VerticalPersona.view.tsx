import * as React from 'react';
import { PersonaCoin } from '../PersonaCoin';
import { Text } from '../Text';
import { IVerticalPersonaComponent } from './VerticalPersona.types';

export const VerticalPersonaView: IVerticalPersonaComponent['view'] = props => {
  const { root, text, secondaryText } = props.classNames;
  const coinProps = props.coinProps || {};

  return (
    <div className={root}>
      <PersonaCoin
        text={props.text}
        size={coinProps.size}
        initials={coinProps.initials}
        imageUrl={coinProps.imageUrl}
        coinColor={coinProps.coinColor}
        initialsColor={coinProps.initialsColor}
        presence={coinProps.presence}
        onPhotoLoadingStateChange={coinProps.onPhotoLoadingStateChange}
      />
      <Text wrap className={text}>
        {props.text}
      </Text>
      <Text wrap className={secondaryText}>
        {props.secondaryText}
      </Text>
    </div>
  );
};
