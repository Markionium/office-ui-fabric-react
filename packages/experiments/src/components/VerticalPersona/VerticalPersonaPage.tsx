import * as React from 'react';
import { ExampleCard, IComponentDemoPageProps, ComponentPage, PropertiesTableSet } from '@uifabric/example-app-base';

import { VerticalPersonaExample } from './examples/VerticalPersona.Example';

const VerticalPersonaExampleCode = require('!raw-loader!@uifabric/experiments/src/components/VerticalPersona/examples/VerticalPersona.Example.tsx') as string;

export class VerticalPersonaPage extends React.Component<IComponentDemoPageProps, {}> {
  public render(): JSX.Element {
    return (
      <ComponentPage
        title=" VerticalPersona"
        componentName=" VerticalPersona"
        exampleCards={
          <div>
            <ExampleCard title="VerticalPersona basic usage" code={VerticalPersonaExampleCode}>
              <VerticalPersonaExample />
            </ExampleCard>
          </div>
        }
        propertiesTables={
          <PropertiesTableSet
            sources={[require<string>('!raw-loader!@uifabric/experiments/src/components/VerticalPersona/VerticalPersona.types.ts')]}
          />
        }
        // overview={<div />}
        // bestPractices={<div />}
        dos={
          <div>
            <ul>
              <li>Provide as few props as needed to render the coin</li>
            </ul>
          </div>
        }
        donts={
          <div>
            <ul />
          </div>
        }
        isHeaderVisible={this.props.isHeaderVisible}
      />
    );
  }
}
