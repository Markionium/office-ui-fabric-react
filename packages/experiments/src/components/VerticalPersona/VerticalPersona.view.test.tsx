import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { IVerticalPersonaStyles } from './VerticalPersona.types';
import { VerticalPersonaView } from './VerticalPersona.view';
import { IProcessedStyleSet } from 'office-ui-fabric-react/lib/Styling';

// These tests will ensure that your styles regions have classname representation in snapshot output.
// (Also, classNames is a required prop for views, so we have to supply it for tests.)
const testVerticalPersonaClassNames: IProcessedStyleSet<IVerticalPersonaStyles> = {
  root: 'test-cn-root',
  text: 'test-cn-text',
  secondaryText: 'test-cn-secondaryText',
  subComponentStyles: {}
};

// Views are just pure functions with no statefulness, which means they can get full code coverage
//    with snapshot tests exercising permutations of the props.
describe('VerticalPersonaView', () => {
  it('renders correctly with only a text', () => {
    const tree = renderer.create(<VerticalPersonaView text="James Bond" classNames={testVerticalPersonaClassNames} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a text and secondary text', () => {
    const tree = renderer
      .create(<VerticalPersonaView text="James Bond" secondaryText="Super secret agent" classNames={testVerticalPersonaClassNames} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the coin with the passed coinProps', () => {
    const tree = renderer
      .create(
        <VerticalPersonaView
          text="James Bond"
          secondaryText="Super secret agent"
          classNames={testVerticalPersonaClassNames}
          coinProps={{ initials: 'MI6', coinColor: 'red' }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
