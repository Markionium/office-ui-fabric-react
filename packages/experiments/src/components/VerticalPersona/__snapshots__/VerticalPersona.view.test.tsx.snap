// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalPersonaView renders correctly with a text and secondary text 1`] = `
<div
  className="test-cn-root"
>
  <div
    className=

        {
          align-items: center;
          background-color: #5E4B8B;
          border-radius: 50%;
          color: white;
          display: flex;
          font-size: 19.2px;
          height: 48px;
          justify-content: center;
          position: relative;
          width: 48px;
        }
  >
    <span
      className=

          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: inherit;
            display: block;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: inherit;
            font-weight: inherit;
            overflow: hidden;
            text-overflow: ellipsis;
            white-space: nowrap;
          }
    >
      JB
    </span>
  </div>
  <span
    className=
        test-cn-text
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: inherit;
          display: block;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: inherit;
          font-weight: inherit;
        }
  >
    James Bond
  </span>
  <span
    className=
        test-cn-secondaryText
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: inherit;
          display: block;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: inherit;
          font-weight: inherit;
        }
  >
    Super secret agent
  </span>
</div>
`;

exports[`VerticalPersonaView renders correctly with only a text 1`] = `
<div
  className="test-cn-root"
>
  <div
    className=

        {
          align-items: center;
          background-color: #5E4B8B;
          border-radius: 50%;
          color: white;
          display: flex;
          font-size: 19.2px;
          height: 48px;
          justify-content: center;
          position: relative;
          width: 48px;
        }
  >
    <span
      className=

          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: inherit;
            display: block;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: inherit;
            font-weight: inherit;
            overflow: hidden;
            text-overflow: ellipsis;
            white-space: nowrap;
          }
    >
      JB
    </span>
  </div>
  <span
    className=
        test-cn-text
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: inherit;
          display: block;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: inherit;
          font-weight: inherit;
        }
  >
    James Bond
  </span>
  <span
    className=
        test-cn-secondaryText
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: inherit;
          display: block;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: inherit;
          font-weight: inherit;
        }
  />
</div>
`;

exports[`VerticalPersonaView renders the coin with the passed coinProps 1`] = `
<div
  className="test-cn-root"
>
  <div
    className=

        {
          align-items: center;
          background-color: red;
          border-radius: 50%;
          color: white;
          display: flex;
          font-size: 19.2px;
          height: 48px;
          justify-content: center;
          position: relative;
          width: 48px;
        }
  >
    <span
      className=

          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: inherit;
            display: block;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: inherit;
            font-weight: inherit;
            overflow: hidden;
            text-overflow: ellipsis;
            white-space: nowrap;
          }
    >
      MI6
    </span>
  </div>
  <span
    className=
        test-cn-text
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: inherit;
          display: block;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: inherit;
          font-weight: inherit;
        }
  >
    James Bond
  </span>
  <span
    className=
        test-cn-secondaryText
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: inherit;
          display: block;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: inherit;
          font-weight: inherit;
        }
  >
    Super secret agent
  </span>
</div>
`;
